### PR TITLE
Add a custom filter chip option to accept avatar

### PR DESCRIPTION
**Pull request by ShadyZekry**
This PR will support #1010 .

- Added a custom option `FormBuilderFilterChipsOption`, to pass to `FormBuilderFilterChip` options instead of passing the generic `FormBuilderFieldOption` to it.
- This will give us the ability to pass a custom `Widget? avatar`.

### CODE REVIEW

File: packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/// in a given menu must represent values with consistent types.
+class FormBuilderFilterChipsOption<T> extends FormBuilderFieldOption<T> {
+  final Widget? avatar;
+
+  /// Creates an option for fields with selection options
+  const FormBuilderFilterChipsOption({
+    Key? key,
+    required value,
+    this.avatar,
+    child,
```
Comment (WilliamCunhaCardoso) [inline packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart:16]:
```suggestion
    super.child,
```
Is it possible to perform it?

File: packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }) : super(
+          key: key,
+          value: value,
+          child: child,
```
Comment (WilliamCunhaCardoso) [inline packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart:20]: Like that. Can you check please? Also for other fields here
```suggestion
    super.child,
  }) : super(
          key: key,
          value: value,
```

**Comment by danvick** [comment]
Thanks for your helpful PR @ShadyZekry.

Can I however request that we have `FormBuilderFilterChipsOption ` renamed to `FormBuilderChipOption`.

This way we can reuse the same class for both `FormBuilderFilterChip` and `FormBuilderFilterChip` as the latter could also benefit from the `avatar` option.

File: packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart (hunk 1 above)
Comment (WilliamCunhaCardoso) [reply packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart:16]: That's an update in dart on latest release. But so far I could not test it. Btw, this is not a required request, I was just asking for a test. Your PR is good for me. The only thing is the owner's request here: https://github.com/danvick/flutter_form_builder/pull/1031#issuecomment-1152160909

**Review by WilliamCunhaCardoso — APPROVED** [review]

File: packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart (hunk 1 above)
Comment (ShadyZekry) [reply packages/flutter_form_builder/lib/src/options/form_builder_filter_chips_option.dart]: Thanks for mentioning, I didn't know that it's possible in newer versions.

**Review by deandreamatias — APPROVED** [review]

**Pull request merged**